Fixed building of HOL Reference under TeXlive 2018

binghe opened the pull request:
Hi,

I got the following errors when trying to build HOL manual PDFs under newly released TeXlive 2018:

```
[471]

! Package inputenc Error: Unicode character ∀ (U+2200)
(inputenc)                not set up for use with LaTeX.

See the inputenc package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.21922    [|- 2 = SUC 1, |- ∀n. SUC n = 1 + n]:
```

The Unicode character (seems encoded in UCS, not even UTF-8) appears inside `verbatim` block, it's not working by TeX default configurations unless `inputenc` package is loaded with Unicode encoding enabled.

Beside this fix, I changed to use new Latin Modern font, and added support of copy&paste 8bit characters from PDF.  Also added the Manual title into PDF metadata.

--Chun

AndreasLoow commented:
Setting inputenc to utf-8 might be fine, but I don't think the other changes should be merged. The font used by all manuals is set in `Manual/LaTeX/layout.sty` (`\usepackage{stix}`). Loading this font also enables T1 fontenc. Lastly, I think loading hyperref the way you do is redundant because it's already loaded, and the title seems to already end up in the metadata of rendered PDFs with the current setup.

mn200 commented:
The issue is that we shouldn’t have that character in the file at all, and it’s good that TL 2018 is being better about reporting the problem.

I will fix the relevant doc file.

Michael

From: Chun Tian <notifications@github.com>
Reply-To: HOL-Theorem-Prover/HOL <reply@reply.github.com>
Date: Friday, 25 May 2018 at 08:29
To: HOL-Theorem-Prover/HOL <HOL@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [HOL-Theorem-Prover/HOL] Fixed building of HOL Reference under TeXlive 2018 (#541)


Hi,

I got the following errors when trying to build HOL manual PDFs under newly released TeXlive 2018:

[471]



! Package inputenc Error: Unicode character ∀ (U+2200)

(inputenc)                not set up for use with LaTeX.



See the inputenc package documentation for explanation.

Type  H <return>  for immediate help.

 ...



l.21922    [|- 2 = SUC 1, |- ∀n. SUC n = 1 + n]:

The Unicode character (seems encoded in UCS, not even UTF-8) appears inside verbatim block, it's not working by TeX default configurations unless inputenc package is loaded with Unicode encoding enabled.

Beside this fix, I changed to use new Latin Modern font, and added support of copy&paste 8bit characters from PDF. Also added the Manual title into PDF metadata.

--Chun

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/HOL-Theorem-Prover/HOL/pull/541

Commit Summary

  *   Fixed building of HOL Reference under TeXlive 2018

File Changes

  *   M Manual/Reference/reference.tex<https://github.com/HOL-Theorem-Prover/HOL/pull/541/files#diff-0> (10)

Patch Links:

  *   https://github.com/HOL-Theorem-Prover/HOL/pull/541.patch
  *   https://github.com/HOL-Theorem-Prover/HOL/pull/541.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/HOL-Theorem-Prover/HOL/pull/541>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAI47yu2d7z0GSRQv0qGxw5_mj6ULRsGks5t1zP_gaJpZM4UNF_j>.

binghe commented:
OK, fair enough, as long as the PDF can be built I’m fine.

Chun

Inviato da iPhone

> Il giorno 25 mag 2018, alle ore 04:30, Michael Norrish <notifications@github.com> ha scritto:
> 
> The issue is that we shouldn’t have that character in the file at all, and it’s good that TL 2018 is being better about reporting the problem. 
> 
> I will fix the relevant doc file. 
> 
> Michael 
> 
> From: Chun Tian <notifications@github.com> 
> Reply-To: HOL-Theorem-Prover/HOL <reply@reply.github.com> 
> Date: Friday, 25 May 2018 at 08:29 
> To: HOL-Theorem-Prover/HOL <HOL@noreply.github.com> 
> Cc: Subscribed <subscribed@noreply.github.com> 
> Subject: [HOL-Theorem-Prover/HOL] Fixed building of HOL Reference under TeXlive 2018 (#541) 
> 
> 
> Hi, 
> 
> I got the following errors when trying to build HOL manual PDFs under newly released TeXlive 2018: 
> 
> [471] 
> 
> 
> 
> ! Package inputenc Error: Unicode character ∀ (U+2200) 
> 
> (inputenc) not set up for use with LaTeX. 
> 
> 
> 
> See the inputenc package documentation for explanation. 
> 
> Type H <return> for immediate help. 
> 
> ... 
> 
> 
> 
> l.21922 [|- 2 = SUC 1, |- ∀n. SUC n = 1 + n]: 
> 
> The Unicode character (seems encoded in UCS, not even UTF-8) appears inside verbatim block, it's not working by TeX default configurations unless inputenc package is loaded with Unicode encoding enabled. 
> 
> Beside this fix, I changed to use new Latin Modern font, and added support of copy&paste 8bit characters from PDF. Also added the Manual title into PDF metadata. 
> 
> --Chun 
> 
> ________________________________ 
> You can view, comment on, or merge this pull request online at: 
> 
> https://github.com/HOL-Theorem-Prover/HOL/pull/541 
> 
> Commit Summary 
> 
> * Fixed building of HOL Reference under TeXlive 2018 
> 
> File Changes 
> 
> * M Manual/Reference/reference.tex<https://github.com/HOL-Theorem-Prover/HOL/pull/541/files#diff-0> (10) 
> 
> Patch Links: 
> 
> * https://github.com/HOL-Theorem-Prover/HOL/pull/541.patch 
> * https://github.com/HOL-Theorem-Prover/HOL/pull/541.diff 
> 
> — 
> You are receiving this because you are subscribed to this thread. 
> Reply to this email directly, view it on GitHub<https://github.com/HOL-Theorem-Prover/HOL/pull/541>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AAI47yu2d7z0GSRQv0qGxw5_mj6ULRsGks5t1zP_gaJpZM4UNF_j>.
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
